packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -1334,6 +1336,182 @@ export class ProjectModel {
                           .returning('*')
                     : [];
 
+            // .dP"Y8    db    Yb    dP 888888 8888b.      .dP"Y8  dP"Yb  88
+            // `Ybo."   dPYb    Yb  dP  88__    8I  Yb     `Ybo." dP   Yb 88
+            // o.`Y8b  dP__Yb    YbdP   88""    8I  dY     o.`Y8b Yb b dP 88  .o
+            // 8bodP' dP""""Yb    YP    888888 8888Y"      8bodP'  `"YoYo 88ood8
+
+            // Get all the saved SQLs
+            const savedSQLs = await trx('saved_sql')
+                .leftJoin('spaces', 'saved_sql.space_uuid', 'spaces.space_uuid')
+                .whereIn('saved_sql.space_uuid', spaceUuids)
+                .andWhere('spaces.project_id', projectId)
+                .select<DbSavedSql[]>('saved_sql.*');
+
+            Logger.info(
+                `Duplicating ${savedSQLs.length} SQL queries on ${previewProjectUuid}`,
+            );
+
+            // Define the type for the new saved SQLs
+            type CloneSavedSQL = InsertSql & {
+                saved_sql_uuid?: string;
+                search_vector?: string;
+            };
+
+            // Create a function to create the saved SQLs
+            const createSavedSQLs = async (savedSQLList: DbSavedSql[]) => {
+                if (savedSQLList.length === 0) {
+                    return [];
+                }
+                // Create an array of promises for generating slugs and mapping saved SQLs
+                const mappedSavedSQLsPromises = savedSQLList.map(async (d) => {
+                    if (!d.space_uuid) {
+                        throw new Error(
+                            `Chart ${d.saved_sql_uuid} has no space_uuid`,
+                        );
+                    }
+                    // Generate the slug asynchronously
+                    const uniqueSlug = await generateUniqueSlug(
```

Comment:
I don't think this will work, since the `generateUniqueSlug` doesn't get unique slugs per project. We need to force these slugs to be the same (in different projects) in order for the promotion to work. (see how we force the slug on spaces)